wrapping/itkBlockMatchingMultiResolutionImageRegistrationMethod.wrap

```diff
@@ -0,0 +1,61 @@
+set(WRAPPER_AUTO_INCLUDE_HEADERS OFF)
+itk_wrap_include("itkPoint.h")
+itk_wrap_include("itkSmartPointer.h")
+itk_wrap_include("itkImage.h")
+itk_wrap_include("itkImageRegion.h")
+
+# Wrap base classes for itk::BlockMatching::MetricImageToDisplacementCalculator public members
+itk_wrap_class("itk::Image" POINTER)
+  foreach(t ${WRAP_ITK_REAL})
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      itk_wrap_template("P${ITKM_${t}}${d}${d}" "itk::Point<${ITKT_${t}},${d}>, ${d}")
+      itk_wrap_template("SPI${ITKM_${t}}${d}${d}" "itk::SmartPointer<itk::Image<${ITKT_${t}},${d}>>, ${d}")
+    endforeach()
+  endforeach()
+itk_end_wrap_class()
+
+itk_wrap_include("itkBlockMatchingMetricImageToDisplacementCalculator.h")
+itk_wrap_class("itk::BlockMatching::MetricImageToDisplacementCalculator" POINTER)
+  foreach(t ${WRAP_ITK_REAL})
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      itk_wrap_template("I${ITKM_${t}}${d}I${ITKM_V${t}${d}}${d}"
+                        "itk::Image<${ITKT_${t}},${d}>, itk::Image<${ITKT_V${t}${d}},${d}>")
+    endforeach()
+  endforeach()
+itk_end_wrap_class()
+
+# Wrap class hierarchy
+itk_wrap_include("itkImageToImageFilter.h")
+itk_wrap_class("itk::ImageToImageFilter" POINTER)
+  foreach(t ${WRAP_ITK_REAL})
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      itk_wrap_template("IIR${d}${d}I${ITKM_V${t}${d}}${d}"
+                        "itk::Image<itk::ImageRegion<${d}>,${d}>, itk::Image<${ITKT_V${t}${d}},${d}>")
+    endforeach()
+  endforeach()
+itk_end_wrap_class()
```

Comment:
Seeing the following warnings when instantiating any BlockMatching class at runtime:
> Template itk::ImageToImageFilter<itk::Image<itk::Vector<double,2>,2>,itk::Image<itk::CovariantVector<double,2>,2>>
 already defined as <class 'itk.itkImageToImageFilterAPython.itkImageToImageFilterIVD22ICVD22'>
 is redefined as {cl}
Warning: template already defined 'itk::ImageToImageFilter<itk::Image<itk::Vector<double,2>,2>,itk::Image<itk::CovariantVector<double,2>,2>>'
Template itk::ImageToImageFilter<itk::Image<itk::Vector<double,3>,3>,itk::Image<itk::CovariantVector<double,3>,3>>
 already defined as <class 'itk.itkImageToImageFilterAPython.itkImageToImageFilterIVD33ICVD33'>
 is redefined as {cl}
Warning: template already defined 'itk::ImageToImageFilter<itk::Image<itk::Vector<double,3>,3>,itk::Image<itk::CovariantVector<double,3>,3>>'

The warnings are introduced with these changes, but we are not explicitly wrapping `ImageToImageFilter` for these templates. Any thoughts on where this could be coming from?

Reply:
:confused: interesting. I do not see anything via inspection, either. :eye:  I will start a local build.

Reply:
@thewtex Were you able to reproduce by any chance?

Reply:
I did not reproduce these warnings locally.

I did get

```
itkBlockMatchingMultiResolutionImageRegistrationMethod: warning(4): ITK type not wrapped, or currently not known: itk::Image< itk::Point< double, 2 >, 2 >
itkBlockMatchingMultiResolutionImageRegistrationMethod: warning(4): ITK type not wrapped, or currently not known: itk::Image< itk::Point< double, 3 >, 3 >
```

at build time.

And,

```
>>> itk.Ultrasound.NormalizedCrossCorrelationMetricImageFilter.New()
Warning: Unknown parameter 'itk::SmartPointer<itk::Image<float,2>>' in template 'itk::Image'
Warning: Unknown parameter 'itk::SmartPointer<itk::Image<float,3>>' in template 'itk::Image'
Namespace already has a value for ImageRegistrationMethod, which is not an itkTemplate instance for class itk::BlockMatching::ImageRegistrationMethod. Overwriting old value.
Namespace already has a value for MultiResolutionImageRegistrationMethod, which is not an itkTemplate instance for class itk::BlockMatching::MultiResolutionImageRegistrationMethod. Overwriting old value.
<itk.ITKCommonBasePython.itkObject; proxy of <Swig Object of type 'itkObject *' at 0x7fa4d95d8660> >
```

at run time.

We could likely replace the `SmartPointer`'s with raw pointers.

Reply:
Interesting, thanks for taking a look. I'll try a clean build over the weekend and see if warnings still appear on my end.

My guess is that the point wrapping warnings are related to my having `ITK_WRAP_DOUBLE` enabled locally, I'll take a closer look.

> We could likely replace the SmartPointer's with raw pointers.
Will do. This would occur within the cxx classes as well as the wrapping files, right?